src/shadow.class.ts

```diff
@@ -1,240 +1,217 @@
+   * @return {String} SVG representation of a shadow
+   */
+  toSVG(object: FabricObject) {
+    const offset = rotateVector(
+        new Point(this.offsetX, this.offsetY),
+        degreesToRadians(-object.angle)
+      ),
+      BLUR_BOX = 20,
+      color = new Color(this.color);
+    let fBoxX = 40,
+      fBoxY = 40;
+
+    if (object.width && object.height) {
+      //http://www.w3.org/TR/SVG/filters.html#FilterEffectsRegion
+      // we add some extra space to filter box to contain the blur ( 20 )
+      fBoxX =
+        toFixed(
+          (Math.abs(offset.x) + this.blur) / object.width,
+          config.NUM_FRACTION_DIGITS
+        ) *
+          100 +
+        BLUR_BOX;
+      fBoxY =
+        toFixed(
+          (Math.abs(offset.y) + this.blur) / object.height,
+          config.NUM_FRACTION_DIGITS
+        ) *
+          100 +
+        BLUR_BOX;
+    }
+    if (object.flipX) {
+      offset.x *= -1;
+    }
+    if (object.flipY) {
+      offset.y *= -1;
+    }
+
+    return `<filter id="SVGID_${this.id}" y="-${fBoxY}%" height="${
+      100 + 2 * fBoxY
+    }%" x="-${fBoxX}%" width="${
+      100 + 2 * fBoxX
+    }%" >\n\t<feGaussianBlur in="SourceAlpha" stdDeviation="${toFixed(
+      this.blur ? this.blur / 2 : 0,
+      config.NUM_FRACTION_DIGITS
+    )}"></feGaussianBlur>\n\t<feOffset dx="${toFixed(
+      offset.x,
+      config.NUM_FRACTION_DIGITS
+    )}" dy="${toFixed(
+      offset.y,
+      config.NUM_FRACTION_DIGITS
+    )}" result="oBlur" ></feOffset>\n\t<feFlood flood-color="${color.toRgb()}" flood-opacity="${color.getAlpha()}"/>\n\t<feComposite in2="oBlur" operator="in" />\n\t<feMerge>\n\t\t<feMergeNode></feMergeNode>\n\t\t<feMergeNode in="SourceGraphic"></feMergeNode>\n\t</feMerge>\n</filter>\n`;
+  }
+
+  /**
+   * Returns object representation of a shadow
+   * @return {Object} Object representation of a shadow instance
+   */
+  toObject() {
+    const data = {
+      color: this.color,
+      blur: this.blur,
+      offsetX: this.offsetX,
+      offsetY: this.offsetY,
+      affectStroke: this.affectStroke,
+      nonScaling: this.nonScaling,
+    };
+
+    if (this.includeDefaultValues) {
+      return data;
+    }
+
+    const defaults = Shadow.prototype;
+    const out: Record<string, unknown> = {};
+    for (const key in data) {
+      if (
+        data[key as keyof typeof data] !== defaults[key as keyof typeof data]
+      ) {
+        out[key] = data[key as keyof typeof data];
+      }
+    }
+    return out;
+  }
 
   /**
    * Regex matching shadow offsetX, offsetY and blur (ex: "2px 2px 10px rgba(0,0,0,0.2)", "rgb(0,255,0) 2px 2px")
-   * @static
-   * @field
-   * @memberOf fabric.Shadow
    */
   // eslint-disable-next-line max-len
-  fabric.Shadow.reOffsetsAndBlur =
+  static reOffsetsAndBlur =
     /(?:\s|^)(-?\d+(?:\.\d*)?(?:px)?(?:\s?|$))?(-?\d+(?:\.\d*)?(?:px)?(?:\s?|$))?(\d+(?:\.\d*)?(?:px)?)?(?:\s?|$)(?:$|\s)/;
-})(typeof exports !== 'undefined' ? exports : window);
```

Comment:
i hate this regex and i wonder if it should exists

Reply:
It can be simplified for sure